### PR TITLE
Take number of arguments at the call-site into account for signature help

### LINKED ISSUE #1065: Signature help appears to be off by one or more parameter
At line https://github.com/rust-analyzer/rust-analyzer/blob/eeb21dc8cc38a0597fb7949b5c1f42408cd93e35/crates/ra_ide_api/src/hover.rs#L77

If I remove the arguments to `reference_definition` and then enter a parenthesis the line becomes: `let ref_result = reference_definition(<|>)` at this point signature help shows up as trying to complete the third argument `name_ref` when it should be pointing at `db`.

This is a regression.

### PR DESCRIPTION
**Pull request by kjeremy**
Fixes #1065

### CODE REVIEW

File: crates/ra_ide_api/src/call_info.rs [hunk 1]
```diff
@@ -49,6 +58,7 @@ pub(crate) fn call_info(db: &RootDatabase, position: FilePosition) -> Option<Cal
                 .args()
                 .position(|arg| arg.syntax().range().contains(position.offset))
                 .or(Some(num_params - 1))
```
Comment (matklad) [inline crates/ra_ide_api/src/call_info.rs]: Let's replace this or with unwrap_or?

File: crates/ra_ide_api/src/call_info.rs [hunk 2]
```diff
@@ -49,6 +58,7 @@ pub(crate) fn call_info(db: &RootDatabase, position: FilePosition) -> Option<Cal
...
+                .min(Some(num_args_of_call))
```
Comment (matklad) [inline crates/ra_ide_api/src/call_info.rs]: Hm, it's unclear why this logic is correct... Perhaps something like `args.take_while(|it| it.range() < position.offset).count()` would work?

Comment (kjeremy) [reply crates/ra_ide_api/src/call_info.rs]: It's basically doing the `min` of the function signature's arguments versus the call site's. I'll see if I can make it more obvious.

Comment (kjeremy) [reply crates/ra_ide_api/src/call_info.rs]: Thanks that cleaned it up quite a bit.

**Pull request merged**